dashboard/scripts/update_under_13_students.rb

```diff
@@ -6,11 +6,16 @@
 puts "Starting to batch update all students under 13."
 num_students_updated = 0
 
+batch_size = 10000
 min_birthday = Date.today - 13.years
-User.where('birthday IS NULL OR birthday > ?', min_birthday).find_each do |user|
-  user.update! sharing_disabled: true
-  num_students_updated += 1
-  puts "Updated #{num_students_updated} students so far." if num_students_updated % 100000 == 0
+User.where('birthday IS NULL OR birthday > ?', min_birthday).in_batches(of: batch_size) do |where|
+  values = where.pluck(:id, :properties)
+  values.each do |_id, properties|
+    properties['sharing_disabled'] = true
+  end
+  User.import([:id, :properties], values, validate: false, on_duplicate_key_update: [:properties])
```

Comment:
@islemaster what does `on_duplicate_key_update` do here?

Reply:
The default mode for the `activerecord-import` gem is a bulk-insert, but here we want to do a bulk-update.  The `on_duplicate_key_update` setting turns this into an upsert operation that will update the `properties` column when on primary keys match (in this case `id`).

See also: [On Duplicate Key Update on the activerecord-import wiki](https://github.com/zdennis/activerecord-import/wiki/On-Duplicate-Key-Update)

Reply:
@joshlory See https://github.com/zdennis/activerecord-import/wiki/On-Duplicate-Key-Update

TL;DR - it's an upsert, when the key (id) already exists update properties